src/Icicle/Dictionary/Parse/Prisms.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+module Icicle.Dictionary.Parse.Prisms (
+    _NTValue
+  , _NTable
+  , _VString
+  , _VInteger
+  , _VFloat
+  , _VBoolean
+  , _VDatetime
+  , _VArray
+  , key
+  ) where
+
+import           Control.Applicative
+import           Control.Lens
+
+import           Data.Bool
+import           Data.Char
+import           Data.Either
+import           Data.Function
+import           Data.Int
+import           Data.Text
+import           Data.Time
+
+import qualified Text.Parsec.Pos as Pos
+
+import           GHC.Float
+
+import           Icicle.Dictionary.Parse.Types
+
+
+_NTValue :: Prism' Node TValue
```

Comment:
incidentally, this was duplicated from here: https://github.com/ambiata/x/blob/master/x-htoml/src/X/Text/Toml.hs

so this was probably faster than doing the template haskell